sdk/core/azure-core/azure/core/pipeline/transport/_base.py

```diff
@@ -519,7 +526,7 @@ def stream_download(self, pipeline: Pipeline[HttpRequest, "HttpResponse"], **kwa
         """
         raise NotImplementedError("stream_download is not implemented.")
 
-    def parts(self) -> Iterator["HttpResponse"]:
+    def parts(self) -> Iterable["HttpResponse"]:
```

Comment:
is this `Iterable` if other overloads of _parts_helper get added later on returning other types?